docs/src/internals.md

```diff
@@ -401,23 +403,29 @@ poly(x, n) = x^n
 const POLY_CONTEXT = Any
 
 # struct for behavior
-struct PolyTerm <: AbstractTerm
-    term::ContinuousTerm
-    deg::Int
+struct PolyTerm{T,D} <: AbstractTerm
+    term::T
+    deg::D
 end
 
 Base.show(io::IO, p::PolyTerm) = print(io, "poly($(p.term), $(p.deg))")
 
-function StatsModels.apply_schema(t::FunctionTerm{typeof(poly)},
+# first pass: create PolyTerm
+StatsModels.apply_schema(t::FunctionTerm{typeof(poly)},
+                         sch::StatsModels.Schema,
+                         Mod::Type{<:POLY_CONTEXT}) =
+    apply_schema(PolyTerm(t.args_parsed...), sch, Mod)
+
+# second pass: apply schema to internal Terms and check for proper types
```

Comment:
Why do we now do this in two passes?

Reply:
I was gonna say, keep reading ;)

Yes a pointer would be appropriate